src/sqlfluff/dialects/dialect_oracle.py

```diff
@@ -50,6 +50,127 @@
 )
 
 
+
+class AlterTableStatementSegment(ansi.AlterTableStatementSegment):
+    """An `ALTER TABLE` statement.
+
+    https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/ALTER-TABLE.html
+    If possible, please keep the order below the same as Oracle's doc:
+    """
+
+    match_grammar: Matchable = Sequence(
+        "ALTER",
+        "TABLE",
+        Ref("TableReferenceSegment"),
+        OneOf(
+            # @TODO all stuff inside this "Delimited" is not validated for Oracle
+            Delimited(
+                OneOf(
+                    # Table options
+                    Sequence(
+                        Ref("ParameterNameSegment"),
+                        Ref("EqualsSegment", optional=True),
+                        OneOf(Ref("LiteralGrammar"), Ref("NakedIdentifierSegment")),
+                    ),
+                    # Add things
+                    Sequence(
+                        OneOf("ADD", "MODIFY"),
+                        Ref.keyword("COLUMN", optional=True),
+                        Ref("ColumnDefinitionSegment"),
+                        OneOf(
+                            Sequence(
+                                OneOf("FIRST", "AFTER"), Ref("ColumnReferenceSegment")
+                            ),
+                            # Bracketed Version of the same
+                            Ref("BracketedColumnReferenceListGrammar"),
+                            optional=True,
+                        ),
+                    ),
+                ),
+            ),
+            Ref("AlterTablePropertiesSegment"),
+            Ref("AlterTableColumnClausesSegment"),
+            Ref("AlterTableConstraintClauses"),
+        ),
+    )
+
+
+
+class AlterTablePropertiesSegment(BaseSegment):
+    """ALTER TABLE `alter_table_properties` per defined in Oracle's grammar.
+
+    https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/ALTER-TABLE.html
+
+    If possible, please match the order of this sequence with what's defined in
+    Oracle's alter_table_properties grammar.
+    """
+
+    type = "alter_table_properties"
+
+    match_grammar = OneOf(
+        # Rename
+        Sequence(
+            "RENAME",
+            "TO",
+            Ref("TableReferenceSegment"),
+        ),
+    )
```

Comment:
Should we add a TODO here to show this is not complete (like you did with the other pieces)? Was a bit confused with your comment above without that.